src/_pytest/config/__init__.py

```diff
@@ -399,7 +399,7 @@ def _getconftestmodules(self, path):
                     mod = self._importconftest(conftestpath)
                     clist.append(mod)
 
-            self._path2confmods[directory] = clist
+            self._path2confmods[path] = clist
```

Comment:
Note that with 1acb18b I've noticed that some other code might expect a dirpath in `self._path2confmods` - haven't checked if I broke this with the previous optimization already though.